datarepo-clienttests/src/main/java/runner/TestRunner.java

```diff
@@ -38,102 +46,87 @@
-      System.out.println("=========================================================");
-      System.out.println("Cleanup After Failure:");
-
       // cleanup deployment (i.e. run teardown method)
-      System.out.println();
-      System.out.println(
-          "Deployment: Calling " + deploymentScript.getClass().getName() + ".deploy()");
+      LOG.info(
+          "Deployment: Calling {}.teardown() after failure", deploymentScript.getClass().getName());
       try {
         if (!config.server.skipDeployment) {
           deploymentScript.teardown();
         }
       } catch (Exception deploymentTeardownEx) {
-        System.out.println("Exception during forced deployment teardown");
-        // deploymentTeardownEx.printStackTrace();
+        LOG.error("Deployment: Exception during forced deployment teardown", deploymentTeardownEx);
       }
 
       // cleanup test scripts (i.e. run cleanup methods)
-      System.out.println();
-      System.out.println("Test Scripts: Calling the cleanup methods");
+      LOG.info("Test Scripts: Calling the cleanup methods after failure");
       try {
         callTestScriptCleanups();
       } catch (Exception testScriptCleanupEx) {
-        System.out.println("Exception during forced test script cleanups");
-        // testScriptCleanupEx.printStackTrace();
+        LOG.error(
+            "Test Scripts: Exception during forced test script cleanups", testScriptCleanupEx);
       }
 
       // cleanup data project (i.e. run cloud cleanup scripts)
+      LOG.info("Data Project: Cleaning up the data project after failure");
       try {
         cleanupDataProject();
       } catch (Exception leftoverTestDataEx) {
-        System.out.println("Exception during forced data project cleanup");
-        // leftoverTestDataEx.printStackTrace();
+        LOG.error("Data Project: Exception during forced data project cleanup", leftoverTestDataEx);
       }
 
       throw originalEx;
     }
   }
 
   void executeTestConfigurationNoGuaranteedCleanup() throws Exception {
-    System.out.println();
-    System.out.println("=========================================================");
-    System.out.println("Deployment: " + (config.server.skipDeployment ? "skipping" : ""));
-
     // specify any value overrides in the Helm chart, then deploy
     if (!config.server.skipDeployment) {
       // get an instance of the deployment script class
       try {
         deploymentScript = config.server.deploymentScript.scriptClass.newInstance();
       } catch (IllegalAccessException | InstantiationException niEx) {
+        LOG.error(
+            "Deployment: Error calling constructor of DeploymentScript class: {}",
+            config.server.deploymentScript.name,
+            niEx);
         throw new IllegalArgumentException(
             "Error calling constructor of DeploymentScript class: "
-                + config.server.deploymentScript.scriptClass.getName(),
+                + config.server.deploymentScript.name,
```

Comment:
Any reason to be inconsistent with using the getter method vs the field?

Reply:
Yes, I thought it was marginally safer. The name field comes from the JSON-formatted TestConfiguration file, while the scriptClass.getName comes from the Java class instantiated by reflection using the aforementioned name property. By this point in the script, the class should already be confirmed, but something went wrong at this point, so I tried to be safer when throwing the exception in hopes of getting a readable error message.